src/prop/cadical.cpp

```diff
@@ -96,12 +96,37 @@ class TimeLimitTerminator : public CaDiCaL::Terminator
   ResourceManager& d_resmgr;
 };
 
+/**
+ * Terminator class that notifies CaDiCaL to terminate when the resource limit
+ * is reached (used for resource limits specified via --rlimit).
+ */
+class ResourceLimitTerminator : public CaDiCaL::Terminator
+{
+ public:
+  ResourceLimitTerminator(ResourceManager& resmgr) : d_resmgr(resmgr){};
+
+  bool terminate() override
+  {
+    d_resmgr.spendResource(Resource::BvSatStep);
+    return d_resmgr.outOfResources();
+  }
+
+ private:
+  ResourceManager& d_resmgr;
+};
```

Comment:
What if we have both an `rlimit` and a `tlimit`? Can't we just combine the two classes above into a single one?

Reply:
Yeah, I guess we could.